### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by enoqueJonas**
# Ruby on Rails capstone project - Budget app

## What does this PR do: 
- Interactions
  - Splash screen
- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
  - name (mandatory)
  - amount (mandatory)
  - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

## Improvements to be made:
- None

### CODE REVIEW

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @enoqueJonas , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights

- Nice UI
- no linters error
- descriptive PR 
- clean and organized code 

## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] As part of the requirements, please make sure to deploy the app and add the link to your readme file
_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 1]
```diff
@@ -1,233 +1,132 @@
...
-
-- **[key_feature_1]**
-- **[key_feature_2]**
-- **[key_feature_3]**
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
-
-<!-- LIVE DEMO -->
-
-## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (brytebee) [inline README.md:110]: Kindly add this section to your README and specify the live page as requested y the previous reviewer.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @enoqueJonas 👋, 

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (BenMukebo) [reply README.md:110]: ##### **Live version**

- [ ] As requested by the previous reviewer, great job adding link to the deployed project in the README. However your link is not correct. Kindly  make sure the project is deployed properly, the home (category )and detail page (transaction) works as expected. here is the page I'm getting:

image

File: app/views/groups/_group.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    </p>
+
+    <p class="group-card-name">
+      <%= group.name %>
+    </p>
+
+    <p class="group-card-amount">
+      $<%= group.transactions_sum %>
+    </p>
+  </a>
```
Comment (BenMukebo) [inline app/views/groups/_group.html.erb:14]: - [OPTIONAL] The card should have a category image, name, `created date`, and amount. kindly let's display the creation date for each category 👇🏻  


Original Design
:------:
 image

File: app/views/groups/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+      <div id="groups">
+        <% @groups.each do |group| %>
+          <%= render group %>
+        <% end %>
+      </div>
+        <a href=<%= new_group_path %> class="wrapper-btn-new-group">
+          <button class="btn-new-group">
+            New Categorie
+          </button>
```
Comment (BenMukebo) [inline app/views/groups/index.html.erb:21]: - [OPTIONAL] Kindly follow the provided design (width, font, size, height, background, color...) Place the "NEW CATEGORY" button at the bottom of your page. ⬇️ 
 The transaction button has a similar problem. Kindly fix them as well. 💯 👍🏻 
- Original
image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi @enoqueJonas, 

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉
- PR has a good title and summary. ✔️
- Gitflow correctly used.  ✔️
- Correctly use Devise for authentication. ✔️ 

##### **Video presentation**

- [OPTIONAL] It's one of the project requiements you receord a video presenting the features of the project and make sure that The maximum length of the video is 5 minutes. Kindly address this 👍🏻 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by enoqueJonas** [comment]
Hello @BenMukebo, @brytebee , @m05e5. I have been trying too deploy my app for hours now. I understand that it's a project requirement, but I only have a few more hours to submit this or I fail the module. I tried different platforms for deployment and it's not working. Can't we get to a solution where I don't fail the module just because of the deployment? The other project requirements are done properly, it's just the deployment.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @enoqueJonas, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did superb work👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**